main6.py

```diff
@@ -0,0 +1,215 @@
+from datetime import datetime
+
+import pandas as pd
+from fastapi import FastAPI, Request
+from fastapi.responses import HTMLResponse
+from fastapi.templating import Jinja2Templates
+from pydantic import BaseModel, Field
+from sqlalchemy import create_engine, Table, Column, Integer, String, ForeignKey, DateTime, MetaData
+
+app = FastAPI()
+
+DATABASE_URL = "sqlite:///./shop.bd"
+
+engine = create_engine(DATABASE_URL)
+
+metadata = MetaData()
+
+templates = Jinja2Templates(directory="templates")
+
+users = Table("users",
+              metadata,
+              Column("id", Integer, primary_key=True),
+              Column("first_name", String(50)),
+              Column("last_name", String(50)),
+              Column("email", String(120)),
+              Column("password", String(50)),
+              )
+
+products = Table(
+    "products",
+    metadata,
+    Column("id", Integer, primary_key=True),
+    Column("name", String(100)),
+    Column("description", String(255)),
+    Column("price", Integer),
+)
+
+orders = Table(
+    "orders",
+    metadata,
+    Column("id", Integer, primary_key=True),
+    Column("user_id", Integer, ForeignKey("users.id")),
+    Column("product_id", Integer, ForeignKey("products.id")),
+    Column("date_ordered", DateTime, default=datetime.utcnow()),
+    Column("status", String, default="Pending"),
+)
+
+
+class User(BaseModel):
+    id: int
+    first_name: str = Field(max_length=50)
+    last_name: str = Field(max_length=50)
+    email: str = Field(max_length=120)
+    password: str = Field(max_length=50)
+
+
+class Product(BaseModel):
+    id: int
+    name: str = Field(max_length=100)
+    description: str = Field(max_length=255)
+    price: float
+
+
+class Order(BaseModel):
+    id: int
+    user_id: int
+    product_id: int
+    date_ordered: datetime
+    status: str
+
+
+metadata.create_all(engine)
+
+
+@app.post("/users/", response_model=User)
+async def create_user(user: User):
+    print("user.model_dump():", user.model_dump())
+    with engine.connect() as connection:
+        query = users.insert().values(**user.model_dump())
+        connection.execute(query)
+
+        print("user:", user)
+        return user
+
+
+@app.get("/users/")
+async def read_users(request: Request):
+    with engine.connect() as connection:
+        query = users.select()
+        print(query)
+        users_result = connection.execute(query).fetchall()
+        print(f"Users fetched: {users_result}")
+        user_table = pd.DataFrame(users_result,
+                                  columns=["id", "first_name", "last_name", "email", "password"]).to_html()
+        return HTMLResponse(content=templates.get_template("shop.html").render(request=request, user_table=user_table))
```

Comment:
Проблема в том, что вы не используете асинхронные инструменты для обращения к БД. Используйте модуль `databases` и еще установите `aiosqlite`. Тогда этот код можно написать так
```suggestion
    query = users.select()
    user_table = pd.DataFrame([user for user in await database.fetch_all(query)]).to_html()
    return templates.TemplateResponse('users.html', {'request': request, 'user_table': user_table})
```
Я исправил только здесь. Дальше сделайте самостоятельно, импортируйте нужный пакет и исправьте конечные точки